ui/lib/kv/addon/routes/secret/metadata.js

```diff
@@ -7,31 +7,63 @@ import Route from '@ember/routing/route';
 import { service } from '@ember/service';
 
 export default class KvSecretMetadataRoute extends Route {
-  @service store;
+  @service capabilities;
   @service secretMountPath;
+  @service store;
 
   fetchMetadata(backend, path) {
     return this.store.queryRecord('kv/metadata', { backend, path }).catch((error) => {
       if (error.message === 'Control Group encountered') {
         throw error;
       }
-      return {};
+      return null;
     });
   }
 
+  async fetchCapabilities(backend, path) {
+    const metadataPath = `${backend}/metadata/${path}`;
+    const dataPath = `${backend}/data/${path}`;
+    const resp = await this.capabilities.fetchMultiplePaths([metadataPath, dataPath]);
+
+    const findPerms = (path) => {
+      const model = resp.find((m) => m.id === path);
+      const { canRead, canUpdate, canDelete } = model;
+      return { canRead, canUpdate, canDelete };
+    };
+
+    return {
+      metadata: findPerms(metadataPath),
+      data: findPerms(dataPath),
+    };
+  }
+
   async model() {
-    const backend = this.secretMountPath.currentPath;
```

Comment:
the typical way of grabbing this info has been via the parentModel so updated for consistency

Reply:
is the parent model always loaded/updated regardless of how you navigate to this route? ex: a `visit('this url')` in a test that goes directly to the route and doesn't touch the parent route.

Reply:
Yes, it is!